internal/service/admin/adminservice/interface.go

```diff
@@ -24,7 +52,18 @@ type Interface interface {
 type Config struct {
 	AdminRepository AtomicRepository
 	Tracer          trace.Tracer
-	Logger          *otelzap.Logger
+	/*
+		Angus: Something to consider: By using a concrete logger, not a logger interface, you tightly couple the
```

Comment:
Your points are truly enlightening, and they made me consider the idea of creating our own logger wrapper, which could utilize zap and otel behind the scenes. This way, we could easily switch out the logger if necessary in the future. I've been pondering over the performance aspects as well.  I'm curious about how much of an impact the zero allocation approach of zap actually has. What do you think? I'm all ears for your thoughts and insights! 💡